core/prometheus/src/lib.rs

```diff
@@ -0,0 +1,106 @@
+    Ok(gauge)
+}
+
+pub fn set_gauge(gauge: &Result<IntGauge>, value: u64) {
+    if let Ok(gauge) = gauge {
+        gauge.set(value as i64);
+    }
+}
+
+lazy_static! {
+    static ref HTTP_COUNTER: Counter = register_counter!(opts!(
+        "example_http_requests_total",
+        "Total number of HTTP requests made.",
+        labels! {"handler" => "all",}
+    ))
+    .unwrap();
+    static ref HTTP_BODY_GAUGE: Gauge = register_gauge!(opts!(
+        "example_http_response_size_bytes",
+        "The HTTP response sizes in bytes.",
+        labels! {"handler" => "all",}
+    ))
+    .unwrap();
+    static ref HTTP_REQ_HISTOGRAM: HistogramVec = register_histogram_vec!(
+        "example_http_request_duration_seconds",
+        "The HTTP request latencies in seconds.",
+        &["handler"]
+    )
+    .unwrap();
+}
+
+/// Initializes the metrics context, and starts an HTTP server
+/// to serve metrics.
+pub fn init_prometheus(prometheus_addr: SocketAddr) {
+    //let addr = &std::net::SocketAddr::V4;
+    //let addr = ([127, 0, 0, 1], 9898).into();
+    let addr = prometheus_addr;
+    //let parsed_addr = addr.parse().unwrap();
+    //prometheus::register_int_counter!("meh", "foo");
+    let server = Server::bind(&addr)
+        .serve(|| {
+            // This is the `Service` that will handle the connection.
+            // `service_fn_ok` is a helper to convert a function that
+            // returns a Response into a `Service`.
+            service_fn_ok(move |req: Request<Body>| {
+                HTTP_COUNTER.inc();
+                if req.uri().path() == "/metrics" {
+                    let metric_families = prometheus::gather();
+                    let mut buffer = vec![];
+                    let encoder = TextEncoder::new();
+                    encoder.encode(&metric_families, &mut buffer).unwrap();
+                    HTTP_BODY_GAUGE.set(buffer.len() as f64);
+                    Response::builder()
+                        .status(StatusCode::OK)
+                        .header("Content-Type", encoder.format_type())
+                        .body(Body::from(buffer))
+                        .expect("Error constructing response")
+                } else {
+                    Response::builder()
+                        .status(StatusCode::NOT_FOUND)
+                        .body(Body::from("Not found."))
+                        .expect("Error constructing response")
+                }
+            })
+        })
+        .map_err(|e| error!("server error: {}", e));
+
+    info!("Exporting metrics at http://{}/metrics", addr);
+
+    let mut rt = tokio::runtime::Builder::new()
+        .core_threads(1) // one thread is sufficient
+        .build()
+        .expect("Unable to build metrics exporter tokio runtime");
+
+    std::thread::spawn(move || {
+        rt.spawn(server);
+        rt.shutdown_on_idle().wait().unwrap();
+    });
+}
+
+
+//impl trait<Number> for dyn Header {
+//    type Number = u64;
+//}
```

Comment:
nit: missing newline at EOF

Reply:
Done at 
https://github.com/nodebreaker-carlsagan/substrate/commit/c48bad6a823bfe62043fb0313ddbdc46815ee9a4

Reply:
Looks good @hskang9.  Please push to the current branch so it shows up in the PR UI.

Reply:
Thanks. Once I finish runtime implementation on prometheus, I will add changes here.